packages/contracts-bridge/contracts/BridgeRouter.sol

```diff
@@ -244,16 +329,77 @@ contract BridgeRouter is Version0, Router {
         bytes29 _tokenId,
         bytes29 _action
     ) internal {
+        // tokens will be sent to the specified recipient
+        address _recipient = _action.evmRecipient();
+        // send tokens
+        _creditTokens(_origin, _nonce, _tokenId, _action, _recipient);
+        // dust the recipient with gas tokens
+        _dust(_recipient);
+    }
+
+    /**
+     * @notice Handles an incoming Connext Transfer message.
+     *
+     * Tokens are sent to the Connext Router,
+     * then Connext.reconcile callback is called
+     *
+     * @param _origin The domain of the chain from which the transfer originated
+     * @param _nonce The unique identifier for the message from origin to destination
+     * @param _tokenId The token ID
+     * @param _action The action
+     */
+    function _handleConnextTransfer(
+        uint32 _origin,
+        uint32 _nonce,
+        bytes29 _tokenId,
+        bytes29 _action
+    ) internal {
+        // tokens will be sent to connext router
+        address _recipient = address(connext);
+        // send tokens
+        address _token = _creditTokens(
+            _origin,
+            _nonce,
+            _tokenId,
+            _action,
+            _recipient
+        );
+        // call Connext reconcile callback
+        connext.reconcile(
```

Comment:
this call will potentially revert. This can cause user funds to become permanently unretrievable

Reply:
update: we have checked this with connext. we're going to make this call ExcessivelySafe

Reply:
using `0` for the type everywhere is basically unintended behavior for the lib. it's cleaner to type your typed mem views